### PR TITLE
Ported QuantizationTest to EE2

### PR DESCRIPTION
**Pull request by gcatron**
Summary:Ported QuantizationTest to EE2

Documentation:

Progress on #3239 

Test Plan: verify everything builds, and QuantizationTest runs and passes.

### CODE REVIEW

File: tests/unittests/QuantizationTest.cpp [hunk 1]
```diff
@@ -393,7 +396,7 @@ TEST(Quantization, enableRowwiseQuantizedFullyConnected) {
 
   quantConfig.enableRowwise = true;
   quantConfig.assertAllNodesQuantized = true;
-  quantization::quantizeFunction(F, quantConfig, *EE.getBackend(),
+  quantization::quantizeFunction(F, quantConfig, *backend.get(),
```
Comment (bertmaher) [inline tests/unittests/QuantizationTest.cpp]: I think the `.get()` is redundant here, and you can just do `*backend`.

File: tests/unittests/QuantizationTest.cpp [hunk 2]
```diff
@@ -491,7 +495,7 @@ TEST(Quantization, enableRowwiseQuantizedFullyConnectedSymmetric) {
   quantConfig.schema = quantization::Schema::Symmetric;
   quantConfig.enableRowwise = true;
   quantConfig.assertAllNodesQuantized = true;
-  quantization::quantizeFunction(F, quantConfig, *EE.getBackend(),
+  quantization::quantizeFunction(F, quantConfig, *backend.get(),
```
Comment (bertmaher) [inline tests/unittests/QuantizationTest.cpp]: Same thing here and a few other places.

File: tests/unittests/QuantizationTest.cpp [hunk 3]
```diff
@@ -974,15 +990,16 @@ static Function *createGRUForQuantization(Module *M,
 TEST_P(Operator, end2endGRU) {
   // STEP1 - Generate the first network to record the quantization parameters.
   auto *mod = &profileEE.getModule();
+  auto *modBackend = &backendSpecificEE.getModule();
   PlaceholderBindings bindings;
-  Function *F1 = createGRUForQuantization(mod, bindings, "main");
-  Function *F2 = F1->clone("main2");
```
Comment (bertmaher) [inline tests/unittests/QuantizationTest.cpp]: Is `clone` totally useless now, since there's no way to "inject" a Function into an EE?

Comment (jfix71) [reply tests/unittests/QuantizationTest.cpp]: It's still used elsewhere, e.g. in `cloneFunInsideFun()` where a function is replicated inside itself many times for parallel testing purposes.

Additionally we talked previously about adding an ability to clone a Module, I don't recall the exact reason we decided against it for now, but would be useful there too.

**Pull request closed without merging**